src/betterproto/templates/template.py.j2

```diff
@@ -80,6 +95,29 @@ class {{ message.py_name }}(betterproto.Message):
         {% endfor %}
     {%  endif %}
 
+    {% if output_file.pydantic_dataclasses and message.has_oneof_fields %}
+    @root_validator()
+    def check_oneof(cls, values):
+        meta = cls._betterproto_meta.oneof_field_by_group # type: ignore
+
+        for group, field_set in meta.items():
+            set_fields = [
+                field.name
+                for field in field_set
+                if values[field.name] is not None
+            ]
+            if len(field_set) == 1:
+                pass
+            elif len(set_fields) == 0:
+                raise ValueError(f"Group {group} has no value; all fields are None")
+            elif len(set_fields) > 1:
+                set_fields_str = ", ".join(set_fields)
+                raise ValueError(
+                    f"Group {group} has more than one value; fields {set_fields_str} are not None"
+                )
+
+        return values
```

Comment:
Would it be possible to add this to the Message class as a check and then just have this method call that public method to avoid accessing private API inside of the code and also reduce code duplication?

Reply:
That should work!